src/vs/workbench/contrib/terminal/common/terminal.ts

```diff
@@ -428,6 +429,13 @@ export interface ITerminalInstance {
 	 */
 	dispose(immediate?: boolean): void;
 
+	/**
+	 * Indicates that a consumer of a renderer only terminal is finished with it.
+	 *
+	 * @param result - Optional result code from a from a task process or custom execution.
+	 */
+	finishedWithRenderer(result?: number): void;
```

Comment:
Why do you need this? Isn't the `TerminalRenderer` disposed when the Terminal gets disposed?

Reply:
Because the "terminal" uses the process manager to wait for the process to exit (see  _createProcess() in src\vs\workbench\contrib\terminal\electron-browser\terminalInstance.ts). When the process exits, it sets up the terminal for re-use (i.e. it doesn't dispose the terminal) by calling _onProcessOrExtensionCallbackExit.

Since a custom task execution does not have an associated process, the terminal system needs to know when the task execution is complete so the terminal can be setup for re-use just like as if a process exited.

Honestly, I think if the concept of what a terminal is representing is abstracted away so it does not have knowledge of a process then this implementation would have been much cleaner. After all, a terminal is just an input\output mechanism, and what is driving the I/O should not be known to the terminal system.

Reply:
@Tyriar  - From your comments below, I believe you understand why this is needed now. Resolving this.